router/usecase/pools/routable_cw_alloy_transmuter_pool.go

```diff
@@ -181,10 +181,94 @@ func (r *routableAlloyTransmuterPoolImpl) CalcTokenOutAmt(tokenIn sdk.Coin, toke
 		return osmomath.BigDec{}, domain.ZeroNormalizationFactorError{Denom: tokenOutDenom, PoolId: r.GetId()}
 	}
 
+	// Check static upper rate limiter
+	// We only need to check it for the token in coin since that is the only one that is increased by the current quote.
+	if err := r.checkStaticRateLimiter(tokenIn); err != nil {
+		return osmomath.BigDec{}, err
+	}
+
 	tokenInAmount := osmomath.BigDecFromSDKInt(tokenIn.Amount)
 
 	tokenInNormFactorBig := osmomath.NewBigIntFromBigInt(tokenInNormFactor.BigInt())
 	tokenOutNormFactorBig := osmomath.NewBigIntFromBigInt(tokenOutNormFactor.BigInt())
 
-	return tokenInAmount.MulInt(tokenOutNormFactorBig).QuoInt(tokenInNormFactorBig), nil
+	tokenOutAmount := tokenInAmount.MulInt(tokenOutNormFactorBig).QuoInt(tokenInNormFactorBig)
+
+	return tokenOutAmount, nil
+}
+
+// checkStaticRateLimiter checks the static rate limiter for the token in coin.
+// Note: static rate limit only has an upper limit.
+// Therefore, we only need to validate the token in balance.
+func (r *routableAlloyTransmuterPoolImpl) checkStaticRateLimiter(tokenInCoin sdk.Coin) error {
+	// If no static rate limiter is set, return
+	if len(r.AlloyTransmuterData.RateLimiterConfig.StaticLimiterByDenomMap) == 0 {
+		return nil
+	}
+
+	// Check if the static rate limiter exists for the token in denom updated balance.
+	tokeInStaticLimiter, ok := r.AlloyTransmuterData.RateLimiterConfig.GetStaticLimiter(tokenInCoin.Denom)
+	if !ok {
+		return nil
+	}
+
+	preComputedData := r.AlloyTransmuterData.PreComputedData
+
+	normalizationFactors := preComputedData.NormalizationScalingFactors
+
+	normalizedBalances := make(map[string]osmomath.Int, len(r.AlloyTransmuterData.AssetConfigs))
+	normalizeTotal := osmomath.ZeroInt()
+
+	// Calculate normalized balances
+	for i := 0; i < len(r.AlloyTransmuterData.AssetConfigs); i++ {
+		assetConfig := r.AlloyTransmuterData.AssetConfigs[i]
+
+		assetDenom := assetConfig.Denom
+
+		assetBalance := r.Balances.AmountOf(assetDenom)
```

Comment:
asset configs contains alloyed asset, which has no balance. This will fail. For the sake of calculating normalized total, we can skip alloyed asset.

Reply:
fixed